app.rb

```diff
@@ -0,0 +1,163 @@
+require './person'
+
+require './student'
+
+require './teacher'
+
+require './book'
+
+require './rental'
+
+class App
+  def initialize
+    @persons = []
+
+    @books = []
+  end
+
+  def run
+    actions.each { |key, value| puts "#{key}: #{value}" }
+
+    action = gets.chomp.to_i
+
+    action(action)
+  end
+
+  private
+
+  def add_person
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+
+    type = gets.chomp.to_i
+
+    puts 'Invalide input' if type != 1 && type != 2
+
+    add_student if type == 1 # Create a student if the user entered 1
+
+    add_teacher if type == 2 # Create a teacher if the user entered 2
+  end
+
+  def list_people(show_index: false)
+    @persons.each_with_index do |person, index|
+      result = "[#{person.class.name}] ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+
+      result = "#{index}) " + result if show_index
+
+      puts result
+    end
+  end
+
+  def add_student
+    print 'Enter the name of the student: '
+
+    name = gets.chomp.to_s
+
+    print 'Enter the age of the student: '
+
+    age = gets.chomp.to_i
+
+    print 'Has parent permission? [Y/N]: '
+
+    parent_permission = gets.chomp.to_s.downcase == 'y'
+
+    student = Student.new(age, name, parent_permission: parent_permission)
+
+    @persons << student
+
+    puts "Student with id #{student.id} created"
+  end
+
+  def add_teacher
+    print 'Enter the name of the teacher: '
+
+    name = gets.chomp.to_s
+
+    print 'Enter the age of the teacher: '
+
+    age = gets.chomp.to_i
+
+    print "And teacher's specialization: "
+
+    specilization = gets.chomp.to_s
+
+    teacher = Teacher.new(specilization, age, name)
+
+    @persons << teacher
+
+    puts "Student with id #{teacher.id} created"
+  end
+
+  def list_books(show_index: false)
+    @books.each_with_index do |book, index|
+      result = "Title: \"#{book.title}\", Author: #{book.author}"
+
+      result = " #{index}) " + result if show_index
+
+      puts result
+    end
+  end
+
+  def add_book
+    print 'Enter the name of the book: '
+
+    name = gets.chomp.to_s
+
+    print 'Enter the author of the book: '
+
+    author = gets.chomp.to_s
+
+    book = Book.new(name, author)
+
+    @books << book
+
+    puts 'Book created successfully'
+  end
+
+  def list_rentals_by_person_id
+    puts 'No person entry found' if @persons.empty?
+
+    print 'Enter person ID: '
+
+    id = gets.chomp.to_i
+
+    @persons.each do |person|
+      if person.id == id
+
+        person.rentals.each do |rental|
+          puts "Date: #{rental.date}, Book: \"#{rental.book.title}\" by #{rental.book.author} "
+        end
+
+      else
+
+        puts "No rentals found for person with id #{id}"
+
```

Comment:
- [ ] **[OPTIONAL]** You may want to fix this error message here that is displayed when listing the rental of a person. Apparently, there is a person with the `id` but it still displays the error message.

  ![err](https://user-images.githubusercontent.com/36542986/180280405-f92b50b0-2437-4aff-bce9-263499bd338c.jpg)